configs/projects/agent-runtime.rb

```diff
@@ -0,0 +1,69 @@
+project 'agent-runtime' do |proj|
+  platform = proj.get_platform
+  # TODO: Is this an acceptable way to inherit from multiple possible agent version branches?
+  proj.setting(:agent_branch, ENV['AGENT_BRANCH'] || 'master')
+
+  proj.inherit_settings 'puppet-agent', 'git://github.com/puppetlabs/puppet-agent', proj.settings[:agent_branch]
+  # Set the version based on `git describe`, then prepend the branch name
+  # TODO: (It seems like there must be a better way to name the package - Is this valuable?)
+  proj.version_from_git
+  proj.version "#{proj.settings[:agent_branch]}.#{proj.settings[:package_version]}"
+  proj.generate_archives true
+  proj.generate_packages false
+
+  proj.description "The puppet agent runtime contains third-party components needed for the puppet agent"
+  proj.license "See components"
+  proj.vendor "Puppet, Inc.  <info@puppet.com>"
+  proj.homepage "https://puppet.com"
+
+  proj.setting(:artifactory_url, "https://artifactory.delivery.puppetlabs.net/artifactory")
+  proj.setting(:buildsources_url, "#{proj.artifactory_url}/generic/buildsources")
+
+  # At least on el-7, ruby_bindir should be set to /opt/puppetlabs/puppet/bin/, which is the same as the regular bindir:
+  # This does not seem to arrive here from the agent project's settings.
+  proj.setting(:ruby_bindir, "#{proj.settings[:bindir]}")
+
+  # The in the PDK ruby-2.1.9 component config, settings[:ruby_dir] is used as the ruby install prefix.
+  # In the puppet-agent repo, that same line uses settings[:prefix] instead of settings[:ruby_dir].
+  # Haven't tested whether it's ok to change that for this repo yet; Use this line as a band-aid for now.
+  proj.setting(:ruby_dir, "#{proj.settings[:prefix]}")
+
+  if platform.is_macos?
+    proj.identifier "com.puppetlabs"
+  end
+
+  # What to build?
+  # --------------
+
+  # Common deps
+  proj.component "openssl"
+  proj.component "curl"
+
+  # Ruby and deps
+  proj.component "runtime-agent"
+  proj.component "puppet-ca-bundle"
+  proj.component "ruby-#{proj.ruby_version}"
+  proj.component "augeas" unless platform.is_windows?
+  proj.component "libxml2" unless platform.is_windows?
+  proj.component "libxslt" unless platform.is_windows?
+  proj.component "ruby-augeas" unless platform.is_windows?
+  # We only build ruby-selinux for EL 5-7
+  if platform.name =~ /^el-([567])-.*/ || platform.is_fedora?
+    proj.component "ruby-selinux"
+  end
+
+  # Here's an idea around setting disparate versions for components when their configs don't differ very much;
+  # Or is it better to just make separate component config files for each version?
```

Comment:
I think it's probably better to have separate component configs for each version, I think that's a bit more consistent / straightforward